packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeDefinition.cs

```diff
@@ -1291,7 +1291,7 @@ private static MethodBodyStatement ThrowValidationFailException(ValueExpression
         /// </summary>
         private MethodBodyStatement[] CreateWritePropertiesStatements()
         {
-            var properties = _model.Properties.Concat(_model.CustomCodeView?.Properties.Where(p => p.WireInfo != null) ?? []);
+            var properties = _model.CanonicalView!.Properties;
```

Comment:
we are using this `!` everywhere which I think is not quite a good practice.
See [this comment](https://github.com/microsoft/typespec/pull/4712/files#r1798702607) for details